params/config_extra_test.go

```diff
@@ -0,0 +1,164 @@
+				ByzantiumBlock:      genesisBlock,
+				ConstantinopleBlock: genesisBlock,
+				PetersburgBlock:     genesisBlock,
+				IstanbulBlock:       genesisBlock,
+				MuirGlacierBlock:    genesisBlock,
+				BerlinBlock:         nil,
+				LondonBlock:         nil,
+				ShanghaiTime:        nil,
+				CancunTime:          nil,
+			},
+		},
+		{
+			fork: upgradetest.ApricotPhase2,
+			expected: &ChainConfig{
+				HomesteadBlock:      genesisBlock,
+				DAOForkBlock:        genesisBlock,
+				DAOForkSupport:      true,
+				EIP150Block:         genesisBlock,
+				EIP155Block:         genesisBlock,
+				EIP158Block:         genesisBlock,
+				ByzantiumBlock:      genesisBlock,
+				ConstantinopleBlock: genesisBlock,
+				PetersburgBlock:     genesisBlock,
+				IstanbulBlock:       genesisBlock,
+				MuirGlacierBlock:    genesisBlock,
+				BerlinBlock:         genesisBlock,
+				LondonBlock:         nil,
+				ShanghaiTime:        nil,
+				CancunTime:          nil,
+			},
+		},
+		{
+			fork: upgradetest.ApricotPhase3,
+			expected: &ChainConfig{
+				HomesteadBlock:      genesisBlock,
+				DAOForkBlock:        genesisBlock,
+				DAOForkSupport:      true,
+				EIP150Block:         genesisBlock,
+				EIP155Block:         genesisBlock,
+				EIP158Block:         genesisBlock,
+				ByzantiumBlock:      genesisBlock,
+				ConstantinopleBlock: genesisBlock,
+				PetersburgBlock:     genesisBlock,
+				IstanbulBlock:       genesisBlock,
+				MuirGlacierBlock:    genesisBlock,
+				BerlinBlock:         genesisBlock,
+				LondonBlock:         genesisBlock,
+				ShanghaiTime:        nil,
+				CancunTime:          nil,
+			},
+		},
+		{
+			fork: upgradetest.Durango,
+			expected: &ChainConfig{
+				HomesteadBlock:      genesisBlock,
+				DAOForkBlock:        genesisBlock,
+				DAOForkSupport:      true,
+				EIP150Block:         genesisBlock,
+				EIP155Block:         genesisBlock,
+				EIP158Block:         genesisBlock,
+				ByzantiumBlock:      genesisBlock,
+				ConstantinopleBlock: genesisBlock,
+				PetersburgBlock:     genesisBlock,
+				IstanbulBlock:       genesisBlock,
+				MuirGlacierBlock:    genesisBlock,
+				BerlinBlock:         genesisBlock,
+				LondonBlock:         genesisBlock,
+				ShanghaiTime:        genesisTimestamp,
+				CancunTime:          nil,
+			},
+		},
+		{
+			fork: upgradetest.Etna,
+			expected: &ChainConfig{
+				HomesteadBlock:      genesisBlock,
+				DAOForkBlock:        genesisBlock,
+				DAOForkSupport:      true,
+				EIP150Block:         genesisBlock,
+				EIP155Block:         genesisBlock,
+				EIP158Block:         genesisBlock,
+				ByzantiumBlock:      genesisBlock,
+				ConstantinopleBlock: genesisBlock,
+				PetersburgBlock:     genesisBlock,
+				IstanbulBlock:       genesisBlock,
+				MuirGlacierBlock:    genesisBlock,
+				BerlinBlock:         genesisBlock,
+				LondonBlock:         genesisBlock,
+				ShanghaiTime:        genesisTimestamp,
+				CancunTime:          genesisTimestamp,
```

Comment:
[no action required] just curious, if the cancun upgrade was applied to coreth in etna, then why is cancuntime set to genesistime and not the canonical etnatime? same question for all other block/timestamps in these tests

Reply:
These are for testing - not production. The test configs have upgrades either initially activated (at genesis) or disabled. Configuring custom times is handled separately